examples/pwa/pwa.js

```diff
@@ -206,13 +206,28 @@ class AmpViewer {
     this.container = container;
 
     win.AMP_SHADOW = true;
-    this.ampReadyPromise_ = new Promise(resolve => {
+    const ampReadyPromise = new Promise(resolve => {
       (window.AMP = window.AMP || []).push(resolve);
     });
-    this.ampReadyPromise_.then(AMP => {
+    ampReadyPromise.then(AMP => {
       log('AMP LOADED:', AMP);
     });
 
+    const isShadowDomSupported = (
+      Element.prototype.attachShadow ||
+      Element.prototype.createShadowRoot
+    );
+    const shadowDomReadyPromise = isShadowDomSupported ?
+        Promise.resolve() :
+        new Promise(resolve => {
```

Comment:
Include a branch on whether polyfill expected or not so that we can easily test both cases. One option - just do `querySelector('script[src*=webcomponents]')` or such.

Reply:
Done. Let me know if my implementation looks correct. I added a branch for rejection if we don't support shadow DOM and can't find a polyfill. So I trust we need to handle that.